src/mono/mono/metadata/reflection.c

```diff
@@ -1548,7 +1548,7 @@ assembly_name_to_aname (MonoAssemblyName *assembly, char *p)
 	}
 	assembly->name = p;
 	s = p;
-	while (*p && (isalnum (*p) || *p == '.' || *p == '-' || *p == '_' || *p == '$' || *p == '@' || g_ascii_isspace (*p)))
+	while (*p && (*p != ','))
 		p++;
 	if (quoted) {
 		if (*p != '"')
```

Comment:
It looks like the quoted path will fail now?

Reply:
How could I test the quoted path?

Reply:
looks like there are some examples here https://github.com/adamsitnik/runtime/blob/b57b100933e306242ae422b9995b47fb3d778dbd/src/libraries/System.Runtime/tests/System.Reflection.Tests/AssemblyNameTests.cs#L37

Reply:
yea.  @fanyang-mono I would suggest doing two separate loops for `quoted` and `!quoted`.   if it's quoted, just take everything until the next close quote (I'm not sure if there's a possibility of escaping a quote).   if `!quoted`, go until the first comma.

---

I can't actually find anything in the docs about quoted assembly names or about escaping quotes.

https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assemblyname?view=net-8.0#remarks
https://learn.microsoft.com/en-us/dotnet/api/system.type.assemblyqualifiedname?view=net-8.0#remarks

Maybe this is some mono invention, or maybe it's backward compat for .NET Framework? need to check what CoreCLR does

Reply:
Also, just for clarity if you really want to walk character by character `p += g_utf8_jump_table[*p];` is how you could do it here or with more validation like https://github.com/adamsitnik/runtime/blob/4246ba19bd196c5f374d94e5c1fc7b21d53bd9fc/src/mono/mono/eglib/gutf8.c#L106-L113

Reply:
looks like CoreCLR these days `AssemblyNameParser` and `TryGetNextToken` does try to handle single and double quotes and escape characters:

https://github.com/dotnet/runtime/blob/1ddfd18d31ece473a1708167a70f92b87e89f008/src/libraries/Common/src/System/Reflection/AssemblyNameParser.cs#L433-L482

Reply:
Would be nice to share this logic somehow.

Reply:
I think we can probably share for most uses in reflection.   The runtime also calls this internally from some places, and also I think the AOT compiler might need it.  One idea is to compile the managed TypeNameParser as a library using NativeAOT and call it from the AOT compiler.